File: python_intro_2/manipulate_the_data.py
import urllib2
import json

"""Parses the json request with a loop from 0 to 29 on the index"""
url = 'https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc'

request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token 3e0df28fcee55ed9d9bfbfda999ac5428e1ab96c'
}

req = urllib2.Request(url, headers=request_headers)

response = urllib2.urlopen(req)
the_page = response.read()
parsed_json = json.loads(the_page)

for i in range(0, 30):
    print(parsed_json['items'][i]['full_name'])

File: ruby_intro/manipulate_the_data.rb
#!/usr/bin/ruby

require "HTTPClient"
require "net/http"
require "uri"
require "json"

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 3e0df28fcee55ed9d9bfbfda999ac5428e1ab96c'
}

clnt = HTTPClient.new

url = 'https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc'
uri = URI(url)
response = Net::HTTP.get(uri)
obj = JSON.parse(response)

puts obj['items'][0]['full_name']
puts obj['items'].map {|allitems| allitems['full_name']}
